doc/integrator/extend_data_model.rst

```diff
@@ -1,55 +1,246 @@
+    from c2cgeoform.views.abstract_views import ListField
+
+
+    base_schema = GeoFormSchemaNode(UserDetail)
+
+    @view_defaults(match_param='table=title')
+    class LuxDownloadUrlViews(AbstractViews):
+        _list_fields = [
+            _list_field('id'),
+            _list_field('name'),
+        ]
+        _id_field = 'id'
+        _model = Title
+        _base_schema = base_schema
+
+        @view_config(
+            route_name='c2cgeoform_index',
+            renderer='./templates/index.jinja2'
+        )
+        def index(self):
+            return super().index()
+
+        @view_config(
+            route_name='c2cgeoform_grid',
+            renderer='fast_json'
+        )
+        def grid(self):
+            return super().grid()
+
+        @view_config(
+            route_name='c2cgeoform_item',
+            request_method='GET',
+            renderer='./templates/edit.jinja2'
+        )
+        def view(self):
+            return super().edit()
+
+        @view_config(
+            route_name='c2cgeoform_item',
+            request_method='POST',
+            renderer='./templates/edit.jinja2'
+        )
+        def save(self):
+            return super().save()
+
+        @view_config(
+            route_name='c2cgeoform_item',
+            request_method='DELETE',
+            renderer='fast_json'
+        )
+        def delete(self):
+            return super().delete()
+
+        @view_config(
+            route_name='c2cgeoform_item_duplicate',
+            request_method='GET',
+            renderer='./templates/edit.jinja2'
+        )
+        def duplicate(self):
+            return super().duplicate()
+
+and finally in ``geoportal/<package>_geoportal/__init__.py`` replace ``config.scan()`` by:
+
+.. code:: python
+
+    # Add custom table in admin interface, that means re-add all normal table
+    from c2cgeoform.routes import register_models
+    from c2cgeoportal_commons.models.main import (
+        Role, LayerWMS, LayerWMTS, Theme, LayerGroup, Interface, OGCServer,
+        Functionality, RestrictionArea)
+    from c2cgeoportal_commons.models.static import User
+    from geoportailv3_geoportal.models import UserDetail, Title
 
-        def __init__(self, name=u'', description=u''):
-            self.name = name
-            self.description = description
+    register_models(config, (
+        ('themes', Theme),
+        ('layer_groups', LayerGroup),
+        ('layers_wms', LayerWMS),
+        ('layers_wmts', LayerWMTS),
+        ('ogc_servers', OGCServer),
+        ('restriction_areas', RestrictionArea),
+        ('users', User),
+        ('roles', Role),
+        ('functionalities', Functionality),
+        ('interfaces', Interface),
+        ('userdetail', UserDetail),
+        ('title', Title),
+    ))
 
-        def __unicode__ (self):
-            return self.name or u''
+    with PermissionSetter(config):
+      # scan view decorator for adding routes
+      config.scan()
```

Comment:
PermissionSetter not imported.
We want permission only on admin views.
```
with PermissionSetter(config):
    config.scan() = config.scan('<package>.admin.views')
config.scan(ignore='<package>.admin.views')
```

Reply:
Thanks Arnaud, but you probably want to say:
```
with PermissionSetter(config):
    config.scan('<package>.admin.views')
config.scan(ignore='<package>.admin.views')
```
right?

Reply:
Yes right